clevertap/src/main/java/com/rudderstack/android/integrations/clevertap/CleverTapIntegrationFactory.java

```diff
@@ -338,19 +339,23 @@ private Date dateFromString(String date) {
     }
 
     private void handleECommerceEvent(Map<String, Object> eventProperties) {
-        HashMap<String, Object> chargeDetails = new HashMap<>();
-        if (eventProperties.containsKey("revenue")) {
-            chargeDetails.put("Amount", getRevenue(eventProperties.get("revenue")));
-        }
-        if (eventProperties.containsKey("order_id")) {
-            chargeDetails.put("Charged ID", eventProperties.get("order_id"));
+        if (eventProperties == null) {
+            return;
         }
-        for (Map.Entry<String, Object> entry : eventProperties.entrySet()) {
-            if (!entry.getKey().equals("products") && !entry.getKey().equals("revenue") && !entry.getKey().equals("order_id") ) {
-                chargeDetails.put(entry.getKey(), entry.getValue());
+        HashMap<String, Object> chargeDetails = new HashMap<>();
+        ArrayList<HashMap<String, Object>> items = new ArrayList<>();
+        for(Map.Entry<String, Object> entrySet: eventProperties.entrySet()) {
+            if (entrySet.getKey().equals("products")) {
+                if (entrySet.getValue() != null && entrySet.getValue() instanceof List<?>)
```

Comment:
Since we are specifically checking for the `List` type and dropping the `products` value otherwise, I'll suggest logging this message with appropriate information (e.g., Since the products are not of the type list, we are dropping it)

PS: If you think it should be good addition then only address it.

Reply:
No need.